src/schemes/boundary/open_boundary/mirroring.jl

```diff
@@ -220,6 +227,36 @@ function mirror_position(particle_coords, boundary_zone)
     return particle_coords - 2 * dist * boundary_zone.plane_normal
 end
 
+average_velocity!(v, u, system, boundary_zone, semi) = v
+
+function average_velocity!(v, u, system, boundary_zone::BoundaryZone{InFlow}, semi)
+    (; plane_normal, zone_origin, initial_condition) = boundary_zone
+
+    # We only use the extrapolated velocity in the vicinity of the transition region.
+    # Otherwise, if the boundary zone is too large, averaging would be excessively influenced
+    # by the fluid velocity farther away from the boundary.
+    max_dist = initial_condition.particle_spacing
+
+    # This function is executed at every stage, so it is possible for buffer particles to temporarily leave the boundary zone.
+    # Thus, we use `abs()` because buffer particles may be located outside the boundary zone.
+    candidates = findall(x -> abs(dot(x - zone_origin, -plane_normal)) <= max_dist,
```

Comment:
Why do you want to include buffer particles outside the boundary zone, but only if they are at most one particle spacing outside the boundary zone?